api/v1/controllers/collectors.js

```diff
@@ -138,13 +137,31 @@ function getCollector(req, res, next) {
 /**
  * Update the specified collector's config data. If a field is not included in
  * the querybody, that field will not be updated.
+ * Some fields are only writable by the collector itself. So, if any of those
+ * fields are being updated, check that the token provided in request belongs to
+ * a collector.
  *
  * @param {IncomingMessage} req - The request object
  * @param {ServerResponse} res - The response object
  * @param {Function} next - The next middleware function in the stack
  */
 function patchCollector(req, res, next) {
-  doPatch(req, res, next, helper);
+  let verifyCtrToken = false;
+  for (let i = 0; i < fieldsWritableByCollectorOnly.length; i++) {
+    const fieldName = fieldsWritableByCollectorOnly[i];
+    if (req.body.hasOwnProperty(fieldName)) {
+      verifyCtrToken = true;
+      break;
+    }
+  }
+
+  if (verifyCtrToken) { // verify that token belongs to collector
+    return jwtUtil.verifyCollectorToken(req)
+    .then(() => doPatch(req, res, next, helper))
+    .catch((err) => u.handleError(next, err, helper.modelName));
+  } else {
+    return doPatch(req, res, next, helper);
+  }
```

Comment:
the else block here is not necessary.

Reply:
More specifically, the "else" is not necessary, but what's inside the else is still necessary.